source/funkin/graphics/adobeanimate/FlxAtlasSprite.hx

```diff
@@ -66,6 +66,8 @@ class FlxAtlasSprite extends FlxAnimate
       throw 'FlxAtlasSprite not initialized properly. Are you sure the path (${path}) exists?';
     }
 
+    onAnimationComplete.add(cleanupAnimation);
```

Comment:
This causes `cleanupAnimation()` to be called twice sometimes (see line 322).

I want to say I vaguely recall having a good reason for not having `cleanupAnimation` in `onAnimationComplete` but I didn't write any comments on it, it seems. We need to review this thoroughly before accepting it because this change in behavior affects ALL FlxAtlasSprites, and any time anyone makes a PR to fix anything they cause four other regressions.